### ISSUE TITLE
MacosScreenAccessor doesn't seem to work

### ISSUE THREAD

**Issue by ychin** [frarees/typometer#3]
I have been playing with MacosScreenAccessor and I can't get it to work for me. Just for reference, I'm using an M1 MacBook Pro with Retina display, so I'm wondering if display scaling is wrong but I played around with it a little but couldn't get it to work. It wouldn't detect the change after the initial addition and deltion.

Looking at the implementation, I do wonder if we could just use the Java AWT implementation. I saw that you forked the project to add a Mac-specific implementation, but looking at OpenJDK's implementation I think they are using basically the same APIs (CGWindowListCreateImage) (https://github.com/openjdk/jdk/blob/50120396b6cca1219fb5dd42a11e4b29b79bd3bd/src/java.desktop/macosx/native/libawt_lwawt/awt/CRobot.m#L315) so in theory the AWT version is "native".

I do wonder if we could use ScreenCaptureKit though, which is a much newer API and is supposed to give much lower access to do screen capture (but only available on macOS 13). I actually wonder if too-low-level of access would skew the result if it bypasses the final window composition. Incidentally, OBS also moved from `CGWindowListCreateImage`to using ScreenCaptureKit (https://github.com/obsproject/obs-studio/pull/5875).

**Comment by frarees** [comment]
You're right pointing out that the native implementation is using the same API. The AWT implementation is doing a better job at it actually, so I think it's reasonable to simplify this and not have the native implementation, which I removed in 15998c1b59ce10a828340daf2ae53baba5a36202.

I've also added permission checks to prompt for screen recording and control in 16b5a18b9137c6dd846b533589658ac3eb132c7b.
If you have any issue running the benchmark, reset the permissions:

```bash
tccutil reset ScreenCapture com.pavelfatin.typometer
tccutil reset Accessibility com.pavelfatin.typometer
```

I'm running it on a MacBook Pro M3.

Regarding `ScreenCaptureKit`, that's a tough effort. Can't bind to it through JNA as-is, we'd need to create C bindings for it.

If we get in trouble in the future and `CGWindowListCreateImage` gets removed (as it's now deprecated) and we need to transision, I'm expecting that AWT Robot would have caught up and we would get the migration for free.

**Comment by frarees** [comment]
Closing this as resolved via 15998c1b59ce10a828340daf2ae53baba5a36202.
